components/user/UserProfile.js

```diff
@@ -87,6 +88,24 @@ function UserProfile({ BASE_URL, data }) {
         </div>
       )}
 
+      {data.isStatsPublic && (
+        <BasicCards
+          data={[
+            {
+              name: "Rank",
+              current: data?.profilStats?.profile?.rank,
```

Comment:
What do you think about, because `data` will need to work for  `data.isStatsPublic`...

```suggestion
              current: data.profilStats?.profile?.rank,
```

Reply:
It's crashes the page if any value is undefined so I am putting these for precautions